File: wrappers/php/lib/Kendo/Dataviz/UI/ChartArea.php
<?php

namespace Kendo\Dataviz\UI;

/**
* PHP wrapper for the chartArea option of Chart.
*/
/**
*/
class ChartArea extends \Kendo\SerializableObject {
//>> Properties

    /**
    * The background color of the chart area. Accepts a valid CSS color string, including hex and rgb.
    * @param string $value
    * @return \Kendo\Dataviz\UI\ChartArea
    */
    public function background($value) {
        return $this->setProperty('background', $value);
    }

    /**
    * The border of the chart area.
    * @param \Kendo\Dataviz\UI\ChartAreaBorder|array $value
    * @return \Kendo\Dataviz\UI\ChartArea
    */
    public function border($value) {
        return $this->setProperty('border', $value);
    }

    /**
    * The height of the chart area.
    * @param float $value
    * @return \Kendo\Dataviz\UI\ChartArea
    */
    public function height($value) {
        return $this->setProperty('height', $value);
    }

    /**
    * The margin of the chart area. A numeric value will set all margins.
    * @param float|\Kendo\Dataviz\UI\ChartAreaMargin|array $value
    * @return \Kendo\Dataviz\UI\ChartArea
    */
    public function margin($value) {
        return $this->setProperty('margin', $value);
    }

    /**
    * The background opacity of the chart area. By default the background is opaque.
    * @param float $value
    * @return \Kendo\Dataviz\UI\ChartArea
    */
    public function opacity($value) {
        return $this->setProperty('opacity', $value);
    }

    /**
    * The width of the chart area.
    * @param float $value
    * @return \Kendo\Dataviz\UI\ChartArea
    */
    public function width($value) {
        return $this->setProperty('width', $value);
    }

//<< Properties
}

?>

File: wrappers/mvc/src/Kendo.Mvc/UI/Chart/ChartArea.cs
namespace Kendo.Mvc.UI
{
    public class ChartArea
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ChartArea" /> class.
        /// </summary>
        public ChartArea()
        {
            Margin = new ChartSpacing();
            Border = new ChartElementBorder();
        }

        /// <summary>
        /// Gets or sets the Chart area background.
        /// </summary>
        /// <value>
        /// The Chart area background.
        /// </value>
        public string Background
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the Chart area border.
        /// </summary>
        /// <value>
        /// The Chart area border.
        /// </value>
        public ChartElementBorder Border
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the Chart area margin.
        /// </summary>
        /// <value>
        /// The Chart area margin.
        /// </value>
        public ChartSpacing Margin
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the Chart area width.
        /// </summary>
        /// <value>
        /// The Chart area width.
        /// </value>
        public int? Width
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the Chart area height.
        /// </summary>
        /// <value>
        /// The Chart area height.
        /// </value>
        public int? Height
        {
            get;
            set;
        }

        public IChartSerializer CreateSerializer()
        {
            return new ChartAreaSerializer(this);
        }
    }
}